app/models/badge.rb

```diff
@@ -1,26 +1,52 @@
 class Badge < ApplicationRecord
-  RARIRIES = %i[common rare ultimate legendary].freeze
+  has_many :acquired_badges, class_name: "User::AcquiredBadge", dependent: :destroy
 
-  belongs_to :user
+  RARIRIES = %i[common rare ultimate legendary].freeze
 
-  def self.badge(name, rarity, icon, description)
+  def self.seed(name, rarity, icon, description)
     raise "Incorrect Rarity" unless RARIRIES.include?(rarity)
 
-    @name = name
-    @rarity = rarity
-    @icon = icon
-    @description = description
+    @seed_data = {
+      name: name,
+      rarity: rarity,
+      icon: icon,
+      description: description
+    }
   end
 
-  %i[name rarity icon description].each do |attr|
-    define_method(attr) { self.class.instance_variable_get("@#{attr}") }
+  # Badges are created on-demand, so if a new badge is added
+  # to the code, it will get added to the database on first usage.
```

Comment:
What is the reason for doing it this way instead of adding records manually? Easier to add new badges I guess?

Reply:
Yeah. Just the fact that we'd have a race condition on deploy, and we'd need to have data being added during migrations which isn't an ideal pattern.

Reply:
Makes sense.